src/libs/localization-api/registry/translation.registry.ts

```diff
@@ -0,0 +1,69 @@
+import { Translation, registerTranslation } from '../manager.js';
+import { hasDefaultExport, loadExtension } from '@umbraco-cms/backoffice/extension-api';
+import { UmbBackofficeExtensionRegistry } from '@umbraco-cms/backoffice/extension-registry';
+import { Subject, combineLatest } from '@umbraco-cms/backoffice/external/rxjs';
+
+export type UmbTranslationDictionary = Record<string, unknown>;
+
+export class UmbTranslationRegistry {
+	#registry;
+	#currentLanguage = new Subject<string>();
+
+	constructor(extensionRegistry: UmbBackofficeExtensionRegistry) {
+		this.#registry = extensionRegistry;
+
+		combineLatest([this.#currentLanguage, this.#registry.extensionsOfType('translations')]).subscribe(
+			async ([userCulture, extensions]) => {
+				await Promise.all(
+					extensions
+						.filter((x) => x.meta.culture.toLowerCase() === userCulture)
+						.map(async (extension) => {
+							const innerDictionary: UmbTranslationDictionary = {};
+
+							// If extension contains a dictionary, add it to the inner dictionary.
+							if (extension.meta.translations) {
+								for (const [dictionaryName, dictionary] of Object.entries(extension.meta.translations)) {
+									this.#addOrUpdateDictionary(innerDictionary, dictionaryName, dictionary);
+								}
+							}
+
+							// If extension contains a js file, load it and add the default dictionary to the inner dictionary.
+							const loadedExtension = await loadExtension(extension);
+
+							if (loadedExtension && hasDefaultExport(loadedExtension)) {
+								for (const [dictionaryName, dictionary] of Object.entries(loadedExtension.default)) {
+									this.#addOrUpdateDictionary(innerDictionary, dictionaryName, dictionary);
+								}
+							}
+
+							// Notify subscribers that the inner dictionary has changed.
+							const translation: Translation = {
+								$code: userCulture,
```

Comment:
We do not use the $ in the rest of out code, so I think we should just get rid of it here. I see it as confusing when reading the code.

Reply:
Okay, I understand now. I find the code a bit hard to read. And I'm a little bit allergic to that type of just mixing properties together style.

I guess I need to see how its suppose to be utilized, cause currently there is no test or code using for example the `$dir`.

Would be nice to bring an example of how someone would use that value, if there is such case. Otherwise there is ways we can get rid of these two props.

Reply:
Lets keep it.. it might be useful for someone.. and I don't have better letter than the $